docs/tutorials/add-caching-to-pg-with-redis.rst

```diff
@@ -0,0 +1,174 @@
+Tutorial: Add caching to your PostgreSQL® app using Redis®*
+===========================================================
+
+With any sufficiently complex application, performance becomes a primary concern to optimize for. One of the key metrics for measuring performance of any software is speed of reading and writing from a database. 
+
+Most applications repeatedly store (write) and retrieve (read) some kind of input from the user or other systems. In most cases, the amount of reads far exceeds the amount of writes. 
+
+Imagine, for example, a creating a customer profile for a store from a web form: the customer fills out their name, phone number, and address once and clicks submit. This creates one write to the database. However, during the checkout process, the application potentially reads that data many times: once to calculate shipping costs based on the customer's address, another time to pre-fill the payment details, and a third time to prompt the customer to receive SMS updates on their shipment. That's 3x the read operations, even in a simple example!
+
+As a result of this, improving read performance gives us a far greater increase in overall performance for our work. The tried and tested way to do this is through using a cache. This becomes especially relevant when we start developing applications in the cloud. When we start making our applications highly available – that is, duplicating various services, like our databases in multiple regions globally to manage speed, and within regions themselves, to manage traffic spikes – we suddenly have data that needs to be read and written to internally, by the application itself, to maintain data integrity, in addition to externally, to be served to our users so they can do things.
```

Comment:
Love all of this - it gives good motivation for why we're bothering with this tutorial